### PR TITLE
when settings:*compile-print-types* print definition types to stdout

### PR DESCRIPTION
**Pull request by garlic0x1**
https://github.com/coalton-lang/coalton/issues/1145

### CODE REVIEW

File: src/settings.lisp [hunk 1]
```diff
@@ -55,6 +56,10 @@ Enable release mode either by setting the UNIX environment variable COALTON_ENV
 (declaim (type boolean *emit-type-annotations*))
 (defvar *emit-type-annotations* t)
 
+(declaim (type boolean *compile-print-types*))
+(defvar *compile-print-types* nil
+  "Print types of definitions to stdout on compile.")
```
Comment (stylewarning) [inline src/settings.lisp:61]: nit: "standard output" and not the UNIXism 🤢 "stdout"

File: src/codegen/program.lisp [hunk 2]
```diff
@@ -165,6 +165,14 @@ A function bound here will be called with a keyword category, and one or more ad
                       ,name
                       ,(codegen-expression node nil env)))
 
+  ;; Print types of definitions
+  (when (or *compile-print* settings:*compile-print-types*) 
+    (dolist (binding bindings)
```
Comment (stylewarning) [inline src/codegen/program.lisp:170]: Do we want to print the package prefix of the symbol being defined? Probably not...

**Review by stylewarning — APPROVED** [review]
I have one very tiny nit.

**Comment by eliaslfox** [comment]
It looks like `*compile-print*` is `t` when compiling with `C-c C-c` in emacs. Was the intent of this pr that types should always be printed when developing locally?

**Comment by stylewarning** [comment]
> It looks like `*compile-print*` is `t` when compiling with `C-c C-c` in emacs. Was the intent of this pr that types should always be printed when developing locally?
> 
> 
> 
> cc @stylewarning

It was more to respect the CL standard variable (allowing the user to turn it off through the CL standard variable). 

Personally I've been finding the feature extremely useful when doing development.

**Comment by garlic0x1** [comment]
> It looks like `*compile-print*` is `t` when compiling with `C-c C-c` in emacs. Was the intent of this pr that types should always be printed when developing locally?
> 
> cc @stylewarning

Interesting, in Sly it is nil, I assume you are using Slime?

**Pull request merged**